packages/kbn-apm-synthtrace/src/lib/synthetics/synthetics_synthtrace_es_client.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { ESDocumentWithOperation, SyntheticsMonitorDocument } from '@kbn/apm-synthtrace-client';
+import { pipeline, Readable, Transform } from 'stream';
+import { SynthtraceEsClient, SynthtraceEsClientOptions } from '../shared/base_client';
+import { getSerializeTransform } from '../shared/get_serialize_transform';
+import { Logger } from '../utils/create_logger';
+
+export type SyntheticsSynthtraceEsClientOptions = Omit<SynthtraceEsClientOptions, 'pipeline'>;
+
+export class SyntheticsSynthtraceEsClient extends SynthtraceEsClient<SyntheticsMonitorDocument> {
+  constructor(options: { client: Client; logger: Logger } & SyntheticsSynthtraceEsClientOptions) {
+    super({
+      ...options,
+      pipeline: syntheticsPipeline(),
+    });
+    this.dataStreams = ['synthetics-*-*'];
+  }
+}
+
+function syntheticsPipeline() {
+  return (base: Readable) => {
+    return pipeline(
+      base,
+      getSerializeTransform<SyntheticsMonitorDocument>(),
+      getRoutingTransform(),
+      (err: unknown) => {
+        if (err) {
+          throw err;
+        }
+      }
+    );
+  };
+}
+
+function getRoutingTransform() {
```

Comment:
nit: Not related to your PR, something which i should have done some time back with logs, but as i am seeing it now here, hence requesting

There are 2 things which you can do here

1. `The data_stream.type` can be hardcoded to `synthetics`. Looking at the client, you are already hardcoding the value to `synthetics` and also there is no method exposed to change it, like you do it for `dataset` and `namespace`. This way you can restrict and control the datastream name.
2. Or if you don't want to do that, then i would request, this transform is used in multiple clients, can you please move it [here](https://github.com/elastic/kibana/blob/ad2af69cc28fb2bfb9b4eb19ffb731103353dcb5/packages/kbn-apm-synthtrace/src/lib/shared) if its not too much and change it for this and logs client, if its not too much of an effort.

Reply:
Addressed in [0088e7](https://github.com/elastic/kibana/pull/191846/commits/0088e782fc3feaecd772e07db8b7f278c1520032)